frontend/javascripts/oxalis/view/settings/dataset_settings_view.js

```diff
@@ -178,7 +178,12 @@ class DatasetSettings extends React.PureComponent<DatasetSettingsProps, State> {
   getHistogram = (layerName: string, layer: DatasetLayerConfiguration) => {
     const { intensityRange } = layer;
     let histograms = [
-      { numberOfElements: 256, elementCounts: new Array(256).fill(0), min: 0, max: 255 },
+      {
+        numberOfElements: intensityRange[1] + 1,
+        elementCounts: new Array(intensityRange[1] + 1).fill(0),
```

Comment:
I'm wondering if it makes sense to create a bin for each possible value. For uint16 this might work, but for uint32 it wouldn't (since the elementsCount variable would allocate 4 GB of memory). The bin count could be independent of the value range (i.e., always create X bins and map multiple values of the given range to one bin). What do you think, @MichaelBuessemeyer?

Reply:
I investigated this further: The backend only sends arrays with sampled color values in with a length of 256. Thus the data is already downscaled into buckets for uint16. No further work is needed here, except changing the default histogram at the lines you pointed out above (line 181 ff.)